src/AWS/Orleans.Persistence.DynamoDB/Provider/DynamoDBGrainStorage.cs

```diff
@@ -151,7 +151,7 @@ public async Task ReadStateAsync<T>(string grainType, GrainReference grainRefere
                         GrainType = fields[GRAIN_TYPE_PROPERTY_NAME].S,
                         GrainReference = fields[GRAIN_REFERENCE_PROPERTY_NAME].S,
                         ETag = int.Parse(fields[ETAG_PROPERTY_NAME].N),
-                        BinaryState = fields.ContainsKey(BINARY_STATE_PROPERTY_NAME) ? fields[BINARY_STATE_PROPERTY_NAME].B.ToArray() : null,
+                        BinaryState = fields.ContainsKey(BINARY_STATE_PROPERTY_NAME) ? fields[BINARY_STATE_PROPERTY_NAME].B?.ToArray() : null,
                         StringState = fields.ContainsKey(STRING_STATE_PROPERTY_NAME) ? fields[STRING_STATE_PROPERTY_NAME].S : string.Empty
```

Comment:
Should we assign null if the key does not exist?
That way we keep consistent with line 391 (`entity.StringState = null;`)

Reply:
Good idea